### PR TITLE
CUDOS-608 CUDOS-622

### PR DESCRIPTION
**Pull request by tgntr**
-Implemented getSigners
-Extended contract functionality
-Migrated contract integration tests to jest
-Refactored account-utils
-Improved template examples

### CODE REVIEW

File: packages/blast-utilities/config-utils.js [hunk 1]
```diff
@@ -3,37 +3,22 @@ const process = require('process')
...
-
 function getConfig() {
-  if (!fsExtra.pathExistsSync(configPath)) {
-    throw new BlastError(`Config file was not found! Make sure that blast.config.js exists at ${configPath}`)
+  const CONFIG_PATH = path.join(process.cwd(), 'blast.config.js')
+  if (!fsExtra.pathExistsSync(CONFIG_PATH)) {
+    throw new BlastError(`Config file was not found! Make sure that blast.config.js exists at ${CONFIG_PATH}`)
   }
-  config = require(configPath)
+  config = require(CONFIG_PATH)
```
Comment (aemil145) [inline packages/blast-utilities/config-utils.js]: can you define this `config`?

Comment (aemil145) [reply packages/blast-utilities/config-utils.js]: This is coming from code styling that is checked by ESLint. There are similar styling errors in `global-functons.js`

**Review by aemil145 — APPROVED** [review]

File: packages/blast-utilities/contract-utils.js [hunk 2]
```diff
@@ -4,97 +4,80 @@ const {
...
-        throw new BlastError(`Contract with name ${this.contractname} was not found, did you compile it?`)
-      }
+  async deploy(initMsg, owner = this.#owner, label = this.#contractName) {
+    if (this.#isDeployed()) {
+      throw new BlastError(`Contract is already deployed!`)
     }
+    
+    this.#owner = owner
+    const uploadReceipt = await this.#uploadContract()
+    console.log(uploadReceipt)
```
Comment (maptuhec) [inline packages/blast-utilities/contract-utils.js]: do we need the console. logs?

File: packages/blast-utilities/contract-utils.js [hunk 3]
```diff
@@ -4,97 +4,80 @@ const {
...
+    const ic = await this.#initContract(uploadReceipt.codeId, initMsg, label)
```
Comment (maptuhec) [inline packages/blast-utilities/contract-utils.js]: can we use more meaningful const here like contractInstance or something like that

File: packages/blast-utilities/contract-utils.js [hunk 4]
```diff
@@ -4,97 +4,80 @@ const {
...
+  async #uploadContract() {
     // TODO: pass gasLimit as a param or read it from config
-    const uploadFee = calculateFee(1_500_000, this.gasPrice)
-    const wasm = fs.readFileSync(this.wasmPath)
-
-    return await this.client.upload(
-      this.defaultAccount,
+    const wasm = fs.readFileSync(this.#wasmPath)
+    const uploadFee = calculateFee(1_500_000, this.#gasPrice)
+    console.log(wasm)
```
Comment (maptuhec) [inline packages/blast-utilities/contract-utils.js]: same here

**Review by Holymir — APPROVED** [review]

**Review by aemil145 — APPROVED** [review]

**Review by maptuhec — APPROVED** [review]

**Pull request merged**